src/EventSubscriber/Redirect404Subscriber.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\redirect\EventSubscriber\Redirect404Subscriber.
+ */
+
+namespace Drupal\redirect\EventSubscriber;
+
+use Drupal\Core\Database\Database;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * An EventSubscriber that listens to redirect 404 errors.
+ */
+class Redirect404Subscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[KernelEvents::EXCEPTION][] = array('onKernelException');
```

Comment:
So you have an empty array, which has an array inside it with the string 'onKernelException'? Is the array() still needed?